### PR TITLE
migrate deno_path to deno_std

### PR DESCRIPTION
**Pull request by zhmushan**

### CODE REVIEW

File: path/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2018 Deno
```
Comment (ry) [inline path/LICENSE]: Copyright 2018 the Deno authors.

Please move this file to the root.

File: path/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Deno Path Manipulation Libraries
+
+Usage:
+
+```ts
+import * as path from 'https://deno.land/x/path/index.ts'
```
Comment (ry) [inline path/README.md]: Great thanks!

File: path/LICENSE (hunk 1 above)
Comment (zhmushan) [reply path/LICENSE]: I am done.

File: path/test/test-path-basename.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
```
Comment (ry) [inline path/test/test-path-basename.ts]: Please rename this file to //path/basename_test.ts
And please add it to //test.ts

File: path/test/test-path-dirname.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
+import * as path from '../index'
+
+test(function dirname() {
```
Comment (ry) [inline path/test/test-path-dirname.ts]: Rename this file to path/dirname_test.ts
add to //test.ts

File: path/test/test-path-extname.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
+import * as path from '../index'
+
+const slashRE = /\//g
+
+const pairs = [
```
Comment (ry) [inline path/test/test-path-extname.ts]: Rename too

File: path/test/test-path-isabsolute.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
```
Comment (ry) [inline path/test/test-path-isabsolute.ts]: rename this file too

File: path/test/test-path-join.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
```
Comment (ry) [inline path/test/test-path-join.ts]: rename this file too

File: path/test/test-path-join.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  // status quo, rather than being obviously right
+  [['c:'], 'c:.'],
+  [['c:.'], 'c:.'],
+  [['c:', ''], 'c:.'],
+  [['', 'c:'], 'c:.'],
+  [['c:.', '/'], 'c:.\\'],
+  [['c:.', 'file'], 'c:file'],
+  [['c:', '/'], 'c:\\'],
+  [['c:', 'file'], 'c:\\file']
+]
```
Comment (ry) [inline path/test/test-path-join.ts]: Very nice tests - awesome.

File: path/test/test-path-parse-format.ts [hunk 9]
```diff
@@ -0,0 +1,158 @@
+import { test, assertEqual } from 'https://deno.land/x/testing/testing.ts'
```
Comment (ry) [inline path/test/test-path-parse-format.ts]: Rename.... and so on for all the tests.

File: test.ts [hunk 10]
```diff
@@ -10,9 +10,15 @@ import { runTests, completePromise } from "net/file_server_test.ts";
 const fileServer = run({
   args: ["deno", "--allow-net", "net/file_server.ts", "."]
 });
+// path test
+const pathTest = run({
+  args: ["deno", "path/test/index.ts"]
```
Comment (ry) [inline test.ts]: Import them all there, remove path/test/index.ts

Comment (zhmushan) [reply test.ts]: ok

**Comment by zhmushan** [comment]
@ry I think I am ready to be reviewed.

File: path/join_test.ts [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    [[" ", "foo"], " /foo"],
+    [[" ", "."], " "],
+    [[" ", "/"], " /"],
+    [[" ", ""], " "],
+    [["/", "foo"], "/foo"],
+    [["/", "/foo"], "/foo"],
+    [["/", "//foo"], "/foo"],
+    [["/", "", "/foo"], "/foo"],
+    [["", "/", "foo"], "/foo"],
+    [["", "/", "/foo"], "/foo"]
```
Comment (ry) [inline path/join_test.ts]: Are these tests lifted from somewhere? If so we need to cite them.

File: test.ts [hunk 12]
```diff
@@ -10,6 +10,17 @@ import { runTests, completePromise } from "net/file_server_test.ts";
...
+// path test
+import "path/basename_test.ts";
+import "path/dirname_test.ts";
+import "path/extname_test.ts";
+import "path/isabsolute_test.ts";
+import "path/join_test.ts";
+import "path/parse_format_test.ts";
+import "path/relative_test.ts";
+import "path/resolve_test.ts";
+import "path/zero_length_strings_test.ts";
```
Comment (ry) [inline test.ts]: 👍

File: path/join_test.ts (hunk 11 above)
Comment (zhmushan) [reply path/join_test.ts]: These tests are referenced from https://github.com/browserify/path-browserify/blob/master/test/

Comment (ry) [reply path/join_test.ts]: Please add a header to each file that originated from there (all of them?) 
```
// Copyright the Browserify authors. MIT License.
// Ported from https://github.com/browserify/path-browserify/
```

Comment (zhmushan) [reply path/join_test.ts]: Copyright belongs to Node contributes.

File: path/constants.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
+import { platform } from "deno";
```
Comment (ry) [inline path/constants.ts]: license

File: path/index.ts [hunk 14]
```diff
@@ -0,0 +1,1422 @@
+import {
```
Comment (ry) [inline path/index.ts]: license

**Pull request merged**